contracts/hyphen/LiquidityPool.sol

```diff
@@ -388,105 +507,122 @@ contract LiquidityPool is
+        require(tokenAddress != address(0), "25");
 
-        // uint256 amountToTransfer, uint256 lpFee, uint256 transferFeeAmount, uint256 gasFee
-        uint256[4] memory transferDetails = getAmountToTransfer(initialGas, tokenAddress, amount, tokenGasPrice);
+        // Calculate token amount on this chain
+        uint256 tokenDecimals = _getTokenDecimals(tokenAddress);
+        require(tokenDecimals & args.sourceChainDecimals != 0, "48");
+        uint256 amount = _getDestinationChainTokenAmount(
+            args.sourceChainAmount,
+            args.sourceChainDecimals,
+            tokenDecimals
+        );
 
-        liquidityProviders.decreaseCurrentLiquidity(tokenAddress, transferDetails[0]);
+        _verifyExitParams(tokenAddress, amount, args.receiver);
 
-        if (tokenAddress == NATIVE) {
-            (bool success, ) = receiver.call{value: transferDetails[0]}("");
-            require(success, "Native Transfer Failed");
-        } else {
-            SafeERC20Upgradeable.safeTransfer(IERC20Upgradeable(tokenAddress), receiver, transferDetails[0]);
+        (uint256 lpFee, uint256 incentivePoolFee, uint256 transferFeeAmount) = _calculateAndUpdateFeeComponents(
+            tokenAddress,
+            amount
+        );
+
+        // Calculate final amount to transfer
+        uint256 amountToTransfer;
+        require(transferFeeAmount <= amount, "28");
+        unchecked {
+            amountToTransfer = amount - (transferFeeAmount);
+        }
+
+        // Enforce that atleast minAmount (encoded in bytes as hyphenArgs[0]) is transferred to receiver
+        // This behaviour is overridable if they initiate the transaction from the ReclaimerEOA
+        if (args.hyphenArgs.length > 0) {
+            (uint256 minAmount, address reclaimerEoa) = abi.decode(args.hyphenArgs[0], (uint256, address));
+            require(
+                tx.origin == reclaimerEoa ||
+                    amountToTransfer >=
+                    _getDestinationChainTokenAmount(minAmount, args.sourceChainDecimals, tokenDecimals),
+                "49"
+            );
         }
 
+        // Refund the gas fee if user is initiating the transaction
+        if (tx.origin == args.receiver) {
+            amountToTransfer += _getDestinationChainTokenAmount(
+                args.gasFeePaidInTokenAmount,
+                args.sourceChainDecimals,
+                tokenDecimals
+            );
+        }
+
+        // Send funds to user
+        liquidityProviders.decreaseCurrentLiquidity(tokenAddress, amountToTransfer);
+        _releaseFunds(tokenAddress, args.receiver, amountToTransfer);
+
         emit AssetSent(
             tokenAddress,
             amount,
-            transferDetails[0],
-            receiver,
-            depositHash,
-            fromChainId,
-            transferDetails[1],
-            transferDetails[2],
-            transferDetails[3]
+            amountToTransfer,
+            args.receiver,
+            "",
```

Comment:
Why deposit hash is empty here? Can't we make the source chain txn hash as a part of CCMP message itself and pass it along the message??